index.js

```diff
@@ -10,6 +10,36 @@ xmlParser.parseString(fs.readFileSync(__dirname+"/iedoc-core.xml","utf-8"), func
     Command.apiDocuments = new Array(result);
 });
 
+/**
+ * Same as mkdir -p <path>
+ *
+ * @throws on collisions with files and missing permissions.
+ *
+ * @param  {string} path  The path to create. Windows and *NIX paths accepted, as well as absolute and relative ones.
+ * @return {void}
+ */
+function mkdirP(path) {
+  var folders = path.split(/[/\\]+/);
+  path = '';
+
+  while (folders.length) {
+    /* This works for both absolute and relative paths, as split on an absolute path will have resulted in an array with the first bit empty. Safe for absolute Windows paths as well: */
+    path += folders.shift() + '/';
+
+    if (!fs.existsSync(path)) {
+      fs.mkdirSync(path);
+    } else if (!fs.statSync(path).isDirectory()) {
+        throw new Error("Cannot create directory '" + path + "'. File of same name already exists.");
+    }
+  }
+}
+
+exports.setLogger=function (logger){
+    var formatter = require(__dirname+"/JavascriptFormatter");
+    formatter.setLogger(logger);

```

Comment:
The logger here is not checked. The logger may miss error or warn implementation, lead to undefined error.
I suggest to use lodash, ```_.defaults```.
```
_.defaults(logger,{
     log   : function () {},
     debug : function () {},
     info  : function () {},
     warn  : console.warn,
     error : console.error
})
```